internal/ignition/ignition.go

```diff
@@ -666,6 +666,24 @@ func (g *installerGenerator) updateDhcpFiles() error {
 	return nil
 }
 
+func encodeIpv6Contents() string {
+	return fmt.Sprintf("data:,%s", url.PathEscape(common.Ipv6DuidRuntimeConf))
+}
+
+func (g *installerGenerator) addIpv6FileInIgnition(ignition string) error {
+	path := filepath.Join(g.workDir, ignition)
+	config, err := parseIgnitionFile(path)
+	if err != nil {
+		return err
+	}
+	setFileInIgnition(config, "/etc/NetworkManager/conf.d/01-ipv6.conf", encodeIpv6Contents(), false, 0o644)
+	err = writeIgnitionFile(path, config)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
IMO can be replaced with `return  err` or `return writeIgnitionFile(path, config)`